tests/run.go

```diff
@@ -446,6 +446,40 @@ func Run(o *RunOption) {
 				verifyMountsInfo(actualMount, expectedMount)
 			})
 
+			ginkgo.It("should create nested bind mounts within a container", func() {
+				const (
+					outerDir  = "/outer"
+					nestedDir = "/outer/nested"
+				)
+
+				// Create the nested directory on the host
+				hostDirectory := ffs.CreateNestedDir(outerDir)
+				nestedDirectory := ffs.CreateNestedDir(nestedDir)
+				defer ffs.DeleteDirectory(hostDirectory)
+
+				// Directory on host to be mounted at hostDirectory in container
+				tempDir := ffs.CreateTempDir("some_dir")
+				defer ffs.DeleteDirectory(tempDir)
+				// Write a file to the nested directory
+				nestedFilePath := filepath.Join(nestedDirectory, "file1.txt")
+				ffs.WriteFile(nestedFilePath, "test")
+
+				// Mount nested directory first followed by parent directory
+				command.RunWithoutSuccessfulExit(o.BaseOpt, "run", "--rm", "--name", testContainerName,
+					"-v", nestedDirectory+":"+nestedDirectory,
+					"-v", tempDir+":"+hostDirectory,
+					defaultImage, "sh", "-c", "ls "+nestedDirectory)
+
+				// Mount parent directory first followed by nested
+				output := command.StdoutStr(o.BaseOpt, "run", "--rm", "--name", testContainerName2,
+					"-v", tempDir+":"+hostDirectory,
+					"-v", nestedDirectory+":"+nestedDirectory,
+					defaultImage, "sh", "-c", "ls "+nestedDirectory)
+				gomega.Expect(output).Should(gomega.ContainSubstring("file1.txt"))
+
+				// Test with env variable
```

Comment:
Based on the workflow implementation in Finch https://github.com/runfinch/finch/pull/417/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR122, maybe you can just skip the tests if FINCH_DOCKER_COMPAT is not set, instead of setting it in tests?

Reply:
Updated, now have 2 sets of tests, one when FINCH_DOCKER_COMPAT is set and the other when it's not.

Reply:
Sorry but just thought from another perspective. If FINCH_DOCKER_COMPAT is the logic specific to Finch, does it make more sense to add it to Finch repo for now instead of common-tests? 

We can keep this PR open until nerdctl change is released to Finch.

Reply:
We could do that but for other subjects(docker, podman etc) `FINCH_DOCKER_COMPAT ` won't have any impact since they won't set it. We would also have to write a bunch of utility functions for file system operations which already exist in this PR, it may not be worth the effort of doing it as a temporary solution. One thing we could do though is change the name of the environment variable to make it more generic like `DOCKER_COMPAT`, wdyt?

Reply:
I thought it is straightforward to move to Finch repo. If it is complicated, it is fine to me to keep in common-tests.